src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -277,6 +275,12 @@ static constexpr Property<PerformanceMode> performance_mode{"PERFORMANCE_HINT"};
  * usually this value comes from the actual use-case (e.g. number of video-cameras, or other sources of inputs)
  */
 static constexpr Property<uint32_t> num_requests{"PERFORMANCE_HINT_NUM_REQUESTS"};
+
+/**
+ * @brief This key identifies shared pointer to the ov::Model, required for some properties (ov::max_batch_size and
+ * ov::optimal_batch_size)
+ */
+static constexpr Property<std::shared_ptr<ov::Model>> model_ptr{"MODEL_PTR"};
```

Comment:
may be just `model`? Ptr does not add any additional value IMO

CC @vladimir-paramuzov